include/crm/common/results.h

```diff
@@ -325,6 +325,29 @@ enum pcmk_exec_status {
     PCMK_EXEC_MAX = PCMK_EXEC_NO_SECRETS, //!< Maximum value for this enum
 };
 
+/*!
+ * \enum pcmk_result_type_e
+ * \brief Types of Pacemaker result codes
+ *
+ * A particular integer can have different meanings within different Pacemaker
+ * result code families. It may be interpretable within zero, one, or multiple
+ * families.
+ *
+ * These values are useful for specifying how an integer result code should be
+ * interpreted in situations involving a generic integer value. For example, a
+ * function that can process multiple types of result codes might accept an
+ * arbitrary integer argument along with a \c pcmk_result_type_e argument that
+ * specifies how to interpret the integer.
+ */
+typedef enum pcmk_result_type_e {
+    pcmk_result_legacy       = 0,   //!< Legacy API function return code
```

Comment:
Do we need to add any kind of deprecation warning here? I didn't, because this enum value per se isn't deprecated; it signifies that some integer should be interpreted as a deprecated type.

Reply:
No, I don't think that this needs a warning.

Note that unless you actually care about the values, you don't need to assign values to the things in an enum.  The C standard guarantees the first element is 0 and all subsequent elements are one more than the previous (section 6.7.2.2 in C17, if you really care).

Reply:
Thanks! I did know that the standard guarantees that. I didn't know if we had any preference to be explicit; we have a few enums that explicitly assign either the first element or all elements, even though they're just the sequence of consecutive integers starting with 0.

I see there are also a bunch that don't explicitly assign values to members, so I'll drop the assignments here.

Reply:
A good reason to keep the explicit assignments in this particular case is that we'll want to deprecate and drop pcmk_result_legacy after the legacy rc's are dropped, and we'll need the explicit assignments for the remaining values to avoid them changing at that time

Another situation where explicit assignments are useful is when the values are ordered and we might want to insert values in the middle someday (in which case the initial values should be something like 10, 20, 30, ...).

Reply:
Ah good point. I'll take care of that